### PR TITLE
Add custom widget example

### PR DESCRIPTION
**Pull request by SylvainCorlay**
I am also cleaning up the loading of custom widgets.

maps

### CODE REVIEW

File: examples/index.html [hunk 1]
```diff
@@ -4,21 +4,23 @@
...
 
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js">
+  </script>
+
   <!-- Configure and load Thebe !-->
   <script type="text/x-thebe-config">
     {
       bootstrap: true,
       binderOptions: {
-        repo: "binder-examples/requirements",
```
Comment (SylvainCorlay) [inline examples/index.html]: TODO: undo this change. ipyleaflet is needed for that example. It should probably be added to `binder-examples/requirements`, or we could make a new `binder-examples/thebelab-demo`.

File: examples/index.html [hunk 2]
```diff
@@ -4,21 +4,23 @@
   <title>Thebe Lab examples</title>
   <link rel="stylesheet" type="text/css" href="index.css" />
 
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js">
+  </script>
```
Comment (SylvainCorlay) [inline examples/index.html]: requirejs is required for loading custom widgets on unpkg.

File: webpack.config.js [hunk 3]
```diff
@@ -97,4 +92,5 @@ module.exports = {
       },
     ],
   },
+  externals: ["'@jupyter-widgets/base'", "'@jupyter-widgets/controls'"]
```
Comment (SylvainCorlay) [inline webpack.config.js]: > So this is really weird. This addition of an external statement causes uglifyjs to complain.
>
> ERROR in index.js from UglifyJs
> Unexpected character '@' [external "@jupyter-widgets/base":1,0][index.js:26225,17]
> When disabling uglify, the webpack stage completes properly, but at runtime I get invalid JS syntax in the generated bundle, namely:
> 
> module.exports = @jupyter-widgets/base;
> which causes the browser to complain: Uncaught SyntaxError: Invalid or unexpected token.
>
> I have tried to understand what is going on there and could not find the issue,

Adding a second set of quotes seems to fix it but I don't know what is going on here.

File: src/thebelab.js [hunk 4]
```diff
@@ -22,12 +22,24 @@ import {
...
 import { Mode } from "@jupyterlab/codemirror";
 
 import "@jupyterlab/theme-light-extension/static/index.css";
 import "@jupyter-widgets/controls/css/widgets.built.css";
 import "./index.css";
 
+import * as base from '@jupyter-widgets/base';
+import * as controls from '@jupyter-widgets/controls';
+
+if (typeof window !== "undefined" && window.require !== "undefined") {
```
Comment (minrk) [inline src/thebelab.js]: check for window.define instead?

**Comment by SylvainCorlay** [comment]
@minrk I think that this is ready to go. It requires publishing `thebelab` `0.2.1` to npmjs.

S.
